tests/e2e/tests/simple/simple1_test.go

```diff
@@ -148,6 +152,82 @@ func TestAuth(t *testing.T) {
 	}
 }
 
+func Test404sDuringChanges(t *testing.T) {
+	url := "http://" + tc.Kube.Ingress + "/fortio/debug"
+	rulePath1 := util.GetResourcePath(routingR1Yaml)
+	rulePath2 := util.GetResourcePath(routingR2Yaml)
+	go func() {
+		time.Sleep(9 * time.Second)
+		glog.Infof("Changing rules mid run to v1/v2")
+		if err := tc.Kube.Istioctl.CreateRule(rulePath1); err != nil {
+			t.Errorf("istioctl rule create %s failed", routingR1Yaml)
+			return
+		}
+		time.Sleep(4 * time.Second)
+		glog.Infof("Changing rules mid run to a/b")
+		if err := tc.Kube.Istioctl.CreateRule(rulePath2); err != nil {
+			t.Errorf("istioctl rule create %s failed", routingR1Yaml)
+			return
+		}
+		time.Sleep(4 * time.Second)
+		util.KubeDelete(tc.Kube.Namespace, rulePath1) // nolint:errcheck
+		util.KubeDelete(tc.Kube.Namespace, rulePath2) // nolint:errcheck
+	}()
+	// run at a low/moderate QPS for a while while changing the routing rules,
+	// check for any non 200s
+	opts := fhttp.HTTPRunnerOptions{
+		RunnerOptions: periodic.RunnerOptions{
+			QPS:        8,
+			Duration:   20 * time.Second,
+			NumThreads: 8,
+		},
+		URL: url,
+	}
+	res, err := fhttp.RunHTTPTest(&opts)
+	if err != nil {
+		t.Fatalf("Generating traffic via fortio failed: %v", err)
+	}
+	numRequests := res.DurationHistogram.Count
+	num200s := res.RetCodes[http.StatusOK]
+	if num200s != numRequests {
```

Comment:
Same as below, should this check for slightly less than numRequests rather than numRequests exactly?

Reply:
it's a bit of the feature/requirement question : should changing rules for existing already deployed/up svc create any error ? in the strictest sense I think it shouldn't - there should be 0 errors, in particular 0 404s